### PR TITLE
Initial checkin of AzuraCast role [for streaming-based radio stations]

### PR DESCRIPTION
**Pull request by m-anish**
Doesn't have checks yet to disable install on rpi

### CODE REVIEW

**Comment by holta** [comment]
> Doesn't have checks yet to disable install on rpi

@m-anish, can you make that happen prior to IIAB 7.0 Preview 1 (might be renamed RC1) expected Saturday June 15?

Also please within ~24h change defaults_vars.yml from:
```
azuracast_install: True
azuracast_enabled: True
```
...to:
```
azuracast_install: False
azuracast_enabled: False
```
Thx!

File: roles/azuracast/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# A full-featured online radio station suite.
+# Works on Ubuntu 18.04, Debian 9, 10. Uses docker
+
+# azuracast_install: False
+# azuracast_enabled: False
+
+# All above are set in: github.com/iiab/iiab/blob/master/vars/default_vars.yml
+# If nec, change them by editing /etc/iiab/local_vars.yml prior to installing!
+
+azuracast_http_port: 8080
```
Comment (holta) [inline roles/azuracast/defaults/main.yml]: Is 8080 really a good idea?  Several IIAB countries/groups use Calibre on port 8080 :/

(@m-anish can you possibly make another suggestions for AzuraCast?)

And document both these AzuraCast ports in https://github.com/iiab/iiab/wiki/IIAB-Networking#list-of-ports--services ?

PS i'm torn as to what to do w.r.t. local_vars_medium.yml and local_vars_big.yml if this does not yet run on Raspberry Pi, but let's discuss various recommendations here over Live Chat if poss.

Comment (m-anish) [reply roles/azuracast/defaults/main.yml]: it can be any port we want. pick one?

Comment (holta) [reply roles/azuracast/defaults/main.yml]: Please also take the initiative to propose an AzuraCast port, that is different from 8080.

Then post these 2 ports to: https://github.com/iiab/iiab/wiki/IIAB-Networking#list-of-ports--services

**Comment by holta** [comment]
Upstream issue raised by @m-anish:

https://github.com/AzuraCast/AzuraCast/issues/1672

**Comment by m-anish** [comment]
This now works for the most part. Just don't install it in the first run of IIAB install, as it can confuse the network detection logic.

File: roles/azuracast/README.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
+==========
+Azuracast README
+==========
+
+This 'azuracast' playbook adds `Azuracast <https://azuracast.com/>`_ to Internet-in-a-Box (IIAB) for network radio station fnctionality. With 'azuracast', one can schedule podcasts, music, and even do live streaming of audio content. A variety of streaming formats are supported.
```
Comment (holta) [inline roles/azuracast/README.rst]:
```suggestion
This 'azuracast' playbook adds `Azuracast <https://azuracast.com/>`_ to Internet-in-a-Box (IIAB) for network radio station functionality. With 'azuracast', one can schedule podcasts, music, and even do live streaming of audio content. A variety of streaming formats are supported.
```

**Comment by holta** [comment]
@m-anish I need 1-or-2 colorful sentences to sell the attraction of AzuraCast for:

http://wiki.laptop.org/go/IIAB/FAQ#What_services_.28IIAB_apps.29_are_suggested_during_installation.3F

_(Got a couple lines here, to help communicate the ideas/value/appeal/ease-of-use of this playbook in non-technical language?)_

**Comment by m-anish** [comment]
@holta how about this?

Azuracast is a self-hosted, all in one radio station platform. One can use azuracast to schedule podcasts, music, and even do live streaming of audio content. A variety of streaming formats are supported.

**Comment by holta** [comment]
> @holta how about this?
> 
> Azuracast is a self-hosted, all in one radio station platform. One can use azuracast to schedule podcasts, music, and even do live streaming of audio content. A variety of streaming formats are supported.

1) Thanks @m-anish: I've cleaned up the punctuation as bit and posted that to http://wiki.laptop.org/go/IIAB/FAQ#What_services_.28IIAB_apps.29_are_suggested_during_installation.3F

2) What should go in the URL column, where I've put `box:????` as a placeholder for now?

3) See `pbx.lan/freepbx` 3 lines below that, in case that too needs to be modified?

4) Please help clean up [roles/azuracast/README.rst](https://github.com/m-anish/iiab/blob/azuracast/roles/azuracast/README.rst) e.g.:
   - spelling of `fnctionality`
   - let's get the capitalization right for `AzuraCast` on this page especially
   - explain or link to any "Getting Started" AzuraCast document (or similar, or video!) for those wanting to see a example radio station 'just work' immediately/quickly, without drowning in endless details

5) After the above are taken care of, is this ready to merge?

**Comment by holta** [comment]
Somewhat Related:

#1743 Add 'Cham' basic online live Videostream capability

File: roles/azuracast/defaults/main.yml (hunk 1 above)
Comment (m-anish) [reply roles/azuracast/defaults/main.yml]: We're now using 10080/10443 for http/https and generally 10000-10100 for azuracast connections.

**Comment by m-anish** [comment]
What plans are there for this to be merged?

I think Jerry's docker PR should be fixed and tested before this goes in perhaps?

**Comment by m-anish** [comment]
With #1765 merged, this should be good to go!

**Comment by holta** [comment]
Refs:

https://github.com/iiab/iiab/issues/1766 AzuraCast fails to install on BIG-sized IIAB on "Debian 10" VM
https://github.com/iiab/iiab/issues/1772 Can we make AzuraCast work on RPi?
https://github.com/AzuraCast/AzuraCast/issues/332 Raspberry Pi 3 (Raspbian) Docker Support
https://github.com/AzuraCast/AzuraCast/issues/1606 Questions about AzuraCast intergration in the internet-in-a-box project

**Pull request merged**